### PR TITLE
initialPaidFee field

### PR DESCRIPTION
**Pull request by miiu96**
- added `initialPaidFee` field in the `transaction` structure
- remove `transactions` and `scresults` in case of rollbacks

### CODE REVIEW

**Review by bogdan-rosianu — APPROVED** [review]

**Review by mariusmihaic — APPROVED** [review]

File: process/elasticProcessor.go [hunk 1]
```diff
@@ -315,14 +315,36 @@ func (ei *elasticProcessor) RemoveMiniblocks(header coreData.HeaderHandler, body
...
+		err := ei.elasticClient.DoQueryRemove(
+			elasticIndexer.ScResultsIndex,
+			converters.PrepareHashesForQueryRemove(encodedScrsHashes),
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	together = append(encodedTxsHashes, encodedScrsHashes...)
```
Comment (bogdan-rosianu) [inline process/elasticProcessor.go:340]: init `together` before this line

File: process/elasticProcessor_test.go [hunk 2]
```diff
@@ -566,7 +566,7 @@ func TestElasticProcessor_RemoveTransactions(t *testing.T) {
 	dbWriter := &mock.DatabaseWriterStub{
 		DoQueryRemoveCalled: func(index string, body *bytes.Buffer) error {
 			bodyStr := body.String()
-			require.Equal(t, elasticIndexer.TransactionsIndex, index)
+			require.True(t, elasticIndexer.TransactionsIndex == index || elasticIndexer.OperationsIndex == index)
```
Comment (bogdan-rosianu) [inline process/elasticProcessor_test.go:569]: `require.Contains(t, []string{elasticIndexer.TransactionsIndex, elasticIndexer.OperationsIndex}, index)`

File: process/elasticProcessor.go (hunk 1 above)
Comment (miiu96) [reply process/elasticProcessor.go:340]: refactored

File: process/elasticProcessor_test.go (hunk 2 above)
Comment (miiu96) [reply process/elasticProcessor_test.go:569]: done.

**Review by bogdan-rosianu — APPROVED** [review]

File: mock/dbTransactionsHandlerStub.go [hunk 3]
```diff
@@ -27,9 +27,9 @@ func (tps *DBTransactionProcessorStub) PrepareTransactionsForDatabase(body *bloc
 	return nil
 }
 
-// GetRewardsTxsHashesHexEncoded -
-func (tps *DBTransactionProcessorStub) GetRewardsTxsHashesHexEncoded(_ coreData.HeaderHandler, _ *block.Body) []string {
-	return nil
+// GetHashesHexEncodedForRemove -
+func (tps *DBTransactionProcessorStub) GetHashesHexEncodedForRemove(_ coreData.HeaderHandler, _ *block.Body) ([]string, []string) {
```
Comment (mariusmihaic) [inline mock/dbTransactionsHandlerStub.go:31]: Grammar correction

```suggestion
func (tps *DBTransactionProcessorStub) GetHexEncodedHashesForRemove(_ coreData.HeaderHandler, _ *block.Body) ([]string, []string) {
```

File: process/elasticProcessor.go [hunk 4]
```diff
@@ -315,14 +315,35 @@ func (ei *elasticProcessor) RemoveMiniblocks(header coreData.HeaderHandler, body
...
+
+	if len(encodedTxsHashes) != 0 {
+		err := ei.elasticClient.DoQueryRemove(
+			elasticIndexer.TransactionsIndex,
+			converters.PrepareHashesForQueryRemove(encodedTxsHashes),
+		)
+		if err != nil {
+			return err
+		}
+	}
```
Comment (mariusmihaic) [inline process/elasticProcessor.go:328]: Could create a func here (smth. like: `removeHashesIfNotEmpty`) with `index` + `[]string hashes` input and reuse below

File: process/elasticProcessor.go [hunk 5]
```diff
@@ -315,14 +315,35 @@ func (ei *elasticProcessor) RemoveMiniblocks(header coreData.HeaderHandler, body
...
+	together := append(encodedTxsHashes, encodedScrsHashes...)
+	if len(together) == 0 {
 		return nil
 	}
 
 	return ei.elasticClient.DoQueryRemove(
-		elasticIndexer.TransactionsIndex,
-		converters.PrepareHashesForQueryRemove(encodedTxsHashes),
+		elasticIndexer.OperationsIndex,
+		converters.PrepareHashesForQueryRemove(together),
```
Comment (mariusmihaic) [inline process/elasticProcessor.go:346]: Same proposed func from above could be reused here too

File: process/transactions/transactionsProcessor.go [hunk 6]
```diff
@@ -162,35 +162,43 @@ func (tdp *txsDatabaseProcessor) setTransactionSearchOrder(transactions map[stri
 	return transactions
 }
 
-// GetRewardsTxsHashesHexEncoded will return reward transactions hashes from body hex encoded
-func (tdp *txsDatabaseProcessor) GetRewardsTxsHashesHexEncoded(header coreData.HeaderHandler, body *block.Body) []string {
+// GetHashesHexEncodedForRemove will return transactions hashes from body hex encoded
```
Comment (mariusmihaic) [inline process/transactions/transactionsProcessor.go:165]:
```suggestion
// GetHashesHexEncodedForRemove will return hex encoded transaction hashes from body 
```

File: process/elasticProcessor.go (hunk 4 above)
Comment (miiu96) [reply process/elasticProcessor.go:328]: done.

File: process/elasticProcessor.go (hunk 5 above)
Comment (miiu96) [reply process/elasticProcessor.go:346]: done.

File: process/transactions/transactionsProcessor.go (hunk 6 above)
Comment (miiu96) [reply process/transactions/transactionsProcessor.go]: done.

File: process/transactions/transactionsProcessor.go [hunk 7]
```diff
@@ -162,8 +162,8 @@ func (tdp *txsDatabaseProcessor) setTransactionSearchOrder(transactions map[stri
 	return transactions
 }
 
-// GetHashesHexEncodedForRemove will return transactions hashes from body hex encoded
-func (tdp *txsDatabaseProcessor) GetHashesHexEncodedForRemove(header coreData.HeaderHandler, body *block.Body) ([]string, []string) {
+// GetHexEncodedHashesForRemove will return hex encoded transactions hashes and smart contract results hashes from body
```
Comment (mariusmihaic) [inline process/transactions/transactionsProcessor.go:165]:
```suggestion
// GetHexEncodedHashesForRemove will return hex encoded transaction hashes and smart contract result hashes from body
```

**Review by mariusmihaic — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**